awx/main/analytics/collectors.py

```diff
@@ -0,0 +1,218 @@
+        counts[cred_type.id] = {'name': cred_type.name,
+                                'credential_count': cred_type.num_credentials
+                                }
+    return counts
+    
+    
+@register('inventory_counts')
+def inventory_counts(since):
+    counts = {}
+    for inv in models.Inventory.objects.filter(kind='').annotate(num_sources=Count('inventory_sources', distinct=True), 
+                                                 num_hosts=Count('hosts', distinct=True)).only('id', 'name', 'kind'):
+        counts[inv.id] = {'name': inv.name,
+                          'kind': inv.kind,
+                          'hosts': inv.num_hosts,
+                          'sources': inv.num_sources
+                          }
+
+    for smart_inv in models.Inventory.objects.filter(kind='smart'):
+        counts[smart_inv.id] = {'name': smart_inv.name,
+                          'kind': smart_inv.kind,
+                          'num_hosts': smart_inv.hosts.count(),
+                          'num_sources': smart_inv.inventory_sources.count()
+                          }
+    return counts
+
+
+@register('projects_by_scm_type')
+def projects_by_scm_type(since):
+    counts = dict(
+        (t[0] or 'manual', 0)
+        for t in models.Project.SCM_TYPE_CHOICES
+    )
+    for result in models.Project.objects.values('scm_type').annotate(
+        count=Count('scm_type')
+    ).order_by('scm_type'):
+        counts[result['scm_type'] or 'manual'] = result['count']
+    return counts
+
+@register('instance_info')   # maybe integrate this with job_instance_counts?
+def instance_info(since):
+    info = {}
+    instances = models.Instance.objects.values_list('hostname').annotate().values('uuid', 'version', 'capacity', 'cpu', 'memory', 'managed_by_policy', 'hostname')
+    for instance in instances:
+        info = {'uuid': instance['uuid'],
+                'version': instance['version'],
+                'capacity': instance['capacity'],
+                'cpu': instance['cpu'],
+                'memory': instance['memory'],
+                'managed_by_policy': instance['managed_by_policy'],
+                }
+    return info
+
+
+@register('job_counts')
+def job_counts(since):
+    counts = {}
+    counts['total_jobs'] = models.UnifiedJob.objects.all().count()
+    counts['status'] = dict(models.UnifiedJob.objects.values_list('status').annotate(Count('status')))
+    counts['launch_type'] = dict(models.UnifiedJob.objects.values_list('launch_type').annotate(Count('launch_type')))
+    
+    return counts
+    
+    
+@register('job_instance_counts')
+def job_instance_counts(since):         #TODO: all of these are going to need to be restrained to the last 24 hours/INSIGHTS_SCHEDULE
+    counts = {}
+
+    job_types = models.UnifiedJob.objects.values_list(
+        'execution_node', 'launch_type').annotate(job_launch_type=Count('launch_type'))
+    for job in job_types:
```

Comment:
For readability's sake, you should probably unpack the variables in the tuple for these loops, something like

`for node, launch_type, count in job_types:`

Reply:
@jbradberry I nested `launch_type` and `status` under each execution node.  Is this what you meant?  For example:

```Out[31]: 
{'awx': 
     {
        'launch_type': {'successful': 6},
        'status': {'dependency': 1, 'manual': 4, 'sync': 1}
      }
}```

Reply:
I just meant that in updating `counts` in the line below, you're referring to `job[0]`, `job[1]`, and `job[2]`.  It'd be nice for future readers of the code for those variables to have names.